engine/crates/engine-v2/src/executor/graphql/query.rs

```diff
@@ -1,37 +1,32 @@
 use std::{collections::HashSet, fmt::Write};
 
 use engine_parser::types::OperationType;
-use itertools::Itertools;
-use schema::{Names, Schema};
 
-use crate::request::{OperationArgument, OperationFields, OperationSelection, OperationSelectionSet, TypeCondition};
+use crate::request::{OperationFieldArgumentWalker, OperationFieldWalker, OperationSelectionSetWalker};
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error(transparent)]
     FmtError(#[from] std::fmt::Error),
 }
 
-pub struct QueryBuilder<'a> {
-    schema: &'a Schema,
-    names: &'a dyn Names,
-    fields: &'a OperationFields,
+pub struct QueryBuilder {
+    buffer: String,
     indent: usize,
+    variable_references: HashSet<String>,
 }
 
-impl<'a> QueryBuilder<'a> {
-    pub fn new(schema: &'a Schema, names: &'a dyn Names, fields: &'a OperationFields) -> Self {
-        Self {
-            schema,
-            names,
-            fields,
+impl QueryBuilder {
```

Comment:
Cleaned up a bit the `QueryBuilder` to use walker and removed the fragment logic. It'll be reworked in the next PR.